src/functions/env.ts

```diff
@@ -49,7 +49,7 @@ const LINE_RE = new RegExp(
   "(" +                      //   begin optional value
   "\\s*'(?:\\\\'|[^'])*'|" + //     single quoted or
   '\\s*"(?:\\\\"|[^"])*"|' + //     double quoted or
-  "[^#\\r\\n]+" +            //     unquoted
+  "[^#=\\r\\n]*" +           //     unquoted
```

Comment:
q: Did we need `=`?

IIUC, we are making something like

```
FOO=bar=a
```

from being a valid entry (`{FOO: "bar=a"}`) to an invalid one.

Reply:
Hmmm, its a bit of a weird case - i guess what we really want to do is allow =, but not after a line break?
So something like ```
[[^#\\r\\n][^#=\\r\\n]]
```?
There may be other regex tricks that could solve this as well - maybe replacing the `\\s` that. matches whitespace after the equals with something that doesn't match newlines?

Reply:
Ahhhh, i just read your original code comments more closely, and realized we explicitly don't support multiline unquoted values. I think i know how to fix this, will play around a bit and make sure all the tests work.